yb-voyager/cmd/exportData_test.go

```diff
@@ -659,3 +749,330 @@ func testCasesWithDifferentTableListFlagValuesTest1(t *testing.T, firstRunTableL
 	assertInitialTableListOnSubsequentRun(t, false, firstRunTableList, firstRunPartitionsToRootMap)
 
 }
+
+func TestTableListInSubsequentRunOfExportDatWithTableListFlagsPGTest2(t *testing.T) {
+	testExportDir := setupPostgreDBAndExportDependencies(t, pgSchemaSqlsTest2, pgSchemasTest2)
+
+	err := testPostgresSource.DB().Connect()
+	if err != nil {
+		utils.ErrExit("Failed to connect to postgres database: %w", err)
+	}
+	defer testPostgresSource.DB().Disconnect()
+	defer testPostgresSource.ExecuteSqls(cleanUpSqlsTest2...)
+	if testExportDir != "" {
+		defer os.RemoveAll(testExportDir)
+	}
+
+	err = InitNameRegistry(testExportDir, SOURCE_DB_EXPORTER_ROLE, testPostgresSource.Source, testPostgresSource.DB(), nil, nil, false)
+	if err != nil {
+		t.Errorf("error initialising name reg for the source: %v", err)
+	}
+
+	metaDB = initMetaDB(testExportDir)
+
+	//Running the command level functions
+	source = *testPostgresSource.Source
+
+	//Fetch table list and partitions to root mapping in the first run
+	expectedPartitionsToRootMapWithoutFlags := map[string]string{
+		`test1.cust_part22`: `test1.customers`,
+		`test1.cust_part21`: `test1.customers`,
+		`test1.cust_part12`: `test1.customers`,
+		`test1.cust_part11`: `test1.customers`,
+		`test2.cust_other`:  `test1.customers`,
+	}
+
+	expectedTableListWithoutFlags := []sqlname.NameTuple{
+		getNameTuple(`test1.cust_part22`),
+		getNameTuple(`test1.cust_part21`),
+		getNameTuple(`test1.cust_part12`),
+		getNameTuple(`test1.cust_part11`),
+		getNameTuple(`test2.cust_other`),
+		getNameTuple(`test1.customers`),
+		getNameTuple(`test1."Foo"`),
+		getNameTuple(`test1."Foo1"`),
+		getNameTuple(`test1."Foo2"`),
+	}
+
+	assertTableListFilteringInTheFirstRun(t, expectedPartitionsToRootMapWithoutFlags, expectedTableListWithoutFlags)
+
+	//case table-list
+	//--table-list test1.customers,test1.Foo* --exclude-table-list test2.cust_arr*
+	source.TableList = "test1.customers,test1.Foo*"
+	source.ExcludeTableList = "test2.cust_arr*"
+
+	//Fetch table list and partitions to root mapping in the first run
+	expectedPartitionsToRootMap := map[string]string{
+		`test2.cust_other`: `test1.customers`,
+	}
+
+	expectedTableList := []sqlname.NameTuple{
+		getNameTuple(`test2.cust_other`),
+		getNameTuple(`test1.customers`),
+		getNameTuple(`test1."Foo"`),
+		getNameTuple(`test1."Foo1"`),
+		getNameTuple(`test1."Foo2"`),
+	}
+
+	assertTableListFilteringInTheFirstRun(t, expectedPartitionsToRootMap, expectedTableList)
+
+	expectedTableListWithOnlyRootTable := []sqlname.NameTuple{}
+
+	//Create msr with required details for subsequent run
```

Comment:
I would have expected that `assertTableListFilteringInTheFirstRun` ultimately calls some function that would also update the state in metadb. Why do you have to manually set that up here to simulate a subsequent run?

Reply:
We update the metadb a little later after the `getInitialTableList`, so I have kept that same as of now. And updating the metadb manually